Add options to QgsDbImportVectorLayerDialog

JanCaha opened the pull request:
## Description

Add options to create spatial index and convert fieldnames to lowercase to QgsDbImportVectorLayerDialog, also adds option to only export selected features. This is to allow actions done in **DB Manage** trough **Import Layer/File** dialog.

Funded by Ocean Winds

JanCaha committed:
add options to create spatial index and convert fieldnames to lowercase on import

JanCaha committed:
add option to only export selected features

nyalldawson commented:
@JanCaha 

A couple of thoughts:

1. I don't think it makes sense anymore to delegate the handling of converting field names to lowercase to the provider methods, as we could now just do that directly in the attributes group with a button that sets all the existing "name" column values to lowercase. This way it would "just work" for all providers, and we'd also be making it nicer for users who want to tweak the field names after converting to lowercase. (Alternatively instead of a button this could be eg a popup menu which shows when right-clicking on the name column, making it easier to add other options like "convert to uppercase" in future)

2. The spatial index support is frustratingly complex. We have a messy situation right now with multiple different APIs for creating spatial indexes, and each has only partial support across a different subset of providers :scream_cat: . Here you've used QgsAbstractDatabaseProviderConnection::createSpatialIndex, which is the more flexible method, but is only supported by gpkg/postgres/spatialite. There's also QgsVectorDataProvider::createSpatialIndex, which is supported by SQL server, and oracle. And then to make things messier, there'll always AUTOMATICALLY be a spatial index created for those providers! So I think before this option is exposed we need to clean this up, so that the options isn't misleading users to think they won't get a spatial index for SQL Server/Oracle when they will. I'd suggest we do this by adding support for QgsAbstractDatabaseProviderConnection::createSpatialIndex to those two providers (as it's the more flexible API), and then take care to disable the automatic QgsVectorLayerExporter creation of spatial index (set mCreateSpatialIndex to false) always when that option is unchecked.

JanCaha committed:
move convert to lowercase into QgsFieldMappingWidget column menu

JanCaha committed:
remove enum

JanCaha committed:
drop options from postgresq

JanCaha committed:
test again connection capabilities

JanCaha committed:
add createSpatialIndex for QgsMssqlProviderConnection

JanCaha commented:
@nyalldawson I was expecting some kind of comment/issues with this :smile: 

1) I removed the checkbox for converting to the lowercase and turned it into context menu for specific column in the **QgsFieldMappingWidget**. It only works when you click on column header. Not sure if this is what you were going for, but it seems reasonable to me. Let me know if this needs to be refined further.

2) That seems reasonable. I dropped parts that should not be required. I also managed to add **createSpatialIndex()** to **QgsMssqlProviderConnection**. That is part that I understand and can verify. But the code for **Oracle** is an issue, as I do not understand it well and cannot quite verify it. Anyways what should be the next steps to properly handle the setting to create the spatial index? Do we add it to **options** of **QgsVectorLayerExporterTask*, add function to set 
**mCreateSpatialIndex** in **QgsVectorLayerExporter**? Or some other way to pass the info?
